span.go

```diff
@@ -90,39 +107,20 @@ type Span interface {
 
 	// Provides access to the Tracer that created this Span.
 	Tracer() Tracer
+
+	// DEPRECATED
```

Comment:
Given that the API we're deprecating was pre-1.0 and has few users, I'd frankly rather just bury it entirely. I want to remove it ASAP but OT has enough users at this point that I can't just delete it. So I will scour the earth for OSS code that references the old API, submit PRs to adopt the new API, then delete all of this.

Inasmuch as removing the docs will help people not-use-it, I would rather remove the docs.